share/eds/byzantine/bad_encoding.go

```diff
@@ -152,6 +156,20 @@ func (p *BadEncodingProof) Validate(hdr libhead.Header) error {
 		)
 	}
 
+	// find at least 1 non-empty share
+	empty := true
+	for _, share := range p.Shares {
+		if share == nil {
+			continue
+		}
+		empty = false
+		break
+	}
+
+	if empty {
+		return errors.New("fraud: invalid proof: no shares provided to reconstruct row/col")
```

Comment:
Hmmm shouldn't you check that at least half the shares are present, instead of at least one?

Reply:
Fixed in [0aafbec](https://github.com/celestiaorg/celestia-node/pull/2383/commits/0aafbec72e7b7bfa2ea84d6375e936aeb1079e63)